library/src/main/scala/scala/scalajs/js/JSConverters.scala

```diff
@@ -51,4 +62,35 @@ object JSConverters {
       arr: scala.Array[T]): JSRichGenTraversableOnce[T] =
     new JSRichGenTraversableOnce(arr)
 
+  @inline
+  implicit def JSRichFutureThenable[A](f: Future[Thenable[A]]): JSRichFuture[A] =
+    new JSRichFuture[A](f.asInstanceOf[Future[A | Thenable[A]]])
```

Comment:
That's why there is a big "Attention unsound!" comment ... These types are only valid when values of `A` don't have a `then` method.

But since a `Thenable[Thenable[A]]` is unsound to begin with, there's no point in dealing with `f: Future[Thenable[Thenable[A]]]`.